### ISSUE TITLE
book 1 chapter 15 typos

### ISSUE THREAD

**Issue by g-i-o-r-g-i-o** [probml/pml-book#672]
sorry if the corrections are very small, but the your book is too great to have big corrections :-D

pag 511
in the section "15.2.6 Vanishing and exploding gradients"

> **Unforunately**, the activations in an RNN can decay or explode as we go **forwards** in time,

--> **Unfortunately**

I'm not sure if **forwards** should be replaced with forward?

---
pag 518

> Q ∈Rm×q are a set of queries (derived from X) used to
> describe what each input is “looking for”, K ∈Rm×q [...] V ∈Rm×v [...] 
> Q = WqX, K = WkX, and V = WvX

---> considering that 
X ∈ Rm×v
and by definition
Q = Wq @ X = (maybe_q × m) @ (m×v)
how can Q ∈ Rm×q? 
If it is a matmul, it should be 
Q = X @ Wq = (m×v) @ (v×q) = (m×q)?

---
pag. 528

15.5.3

> For example, if we assume the sequence length is **n = 3**, we get the following sequence of d = 3-dimensional 

obviously absolutely minor, but was meant to be 

---> **n=8**

since later you show 8 numbers?

---

pag. 534

> ...is conducted only within a block, then space/time complexity is reduced from O(N^2) to **N^2/ K** .

maybe write 
**O(N^2/k)**

---

pag. 535

> to time complexity O(NM 2 log(M)) of the attention module, where M stands for the **dimenionsality** of tokens’ embeddings.

---> **dimensionality**

---
pag. 538

> Figure 15.32: Illustration of ELMo **bidrectional** language model. Here yt = xt+1 when acting as the target

---> **bidirectional**

---

pag.539

15.7.2 BERT

not clear, maybe it's the opposite: 

> The modified input at location t omits all words except for the t’th, and the task is to predict the missing word.

my understanding of BERT is that only the t'th is omitted: In BERT’s masked language modeling, only a subset of tokens (typically about 15%) is masked (replaced with a special [MASK] token), while the rest of the sequence remains intact. 

---

pag 540
15.7.2.3 Fine-tuning BERT for NLP applications

> After pre-training BERT in an unsupervised way, we can use it for various **downtream** tasks by performing supervised fine-tuning.

--> **downstream**

---

**Comment by murphyk** [comment]
thanks, I have fixed these typos in the latest version (will upload soon).
I have also added citations to newer papers that discuss LLMs in more detail.
